sdk/identity/Azure.Identity/samples/DefiningCustomCredentialTypes.md

```diff
@@ -0,0 +1,128 @@
+    {
+        AuthenticationResult result = await _confidentialClient.AcquireTokenForClient(requestContext.Scopes).ExecuteAsync();
+
+        return new AccessToken(result.AccessToken, result.ExpiresOn);
+    }
+}
+```
+
+The users could then use the `ConfidentialClientApplicationCredential` be used to authenticate a `BlobClient` with an `IConfidentialClientApplication` it has created to authenticate non Azure SDK calls.
+
+```C# Snippet:ConfidentialClientCredentialUsage
+IConfidentialClientApplication confidentialClient = ConfidentialClientApplicationBuilder.Create(clientId).WithClientSecret(clientSecret).Build();
+
+var client = new SecretClient(new Uri("https://myvault.vauls.azure.net/"), new ConfidentialClientCredential(confidentialClient));
+```
+
+## Authenticating with the On Behalf Of Flow
+Currently the Azure.Identity library doesn't provide a credential type for clients which need to authenticate with via the On Behalf Of flow. While future support for this is planned, users requiring this immediately will have to implement their own `TokenCredential` class.
+
+In this example the `OnBehalfOfCredential` accepts a client id, client secret, and a user's access token, and creates an instance of `IConfidentialClientApplication` from the Microsoft.Identity.Client library (MSAL) to obtain an obo tokens which can be used to authenticate client requests.
+
+```C# Snippet:OnBehalfOfCredential
+public class OnBehalfOfCredential : TokenCredential
+{
+    private readonly IConfidentialClientApplication _confidentialClient;
+    private readonly UserAssertion _userAssertion;
+
+    public OnBehalfOfCredential(string clientId, string clientSecret, string userAccessToken)
+    {
+        _confidentialClient = ConfidentialClientApplicationBuilder.Create(clientId).WithClientSecret(clientSecret).Build();
+
+        _userAssertion = new UserAssertion(userAccessToken);
+    }
+
+    public override AccessToken GetToken(TokenRequestContext requestContext, CancellationToken cancellationToken)
+    {
+        return GetTokenAsync(requestContext, cancellationToken).GetAwaiter().GetResult();
+    }
+
+    public override async ValueTask<AccessToken> GetTokenAsync(TokenRequestContext requestContext, CancellationToken cancellationToken)
+    {
+        AuthenticationResult result = await _confidentialClient.AcquireTokenOnBehalfOf(requestContext.Scopes, _userAssertion).ExecuteAsync();
+
+        return new AccessToken(result.AccessToken, result.ExpiresOn);
+    }
+}
+```
+The following example shows an how the `OnBehalfOfCredential` could be used to authenticate a `SecretClient`.
+
+
+```C# Snippet:OnBehalfOfCredentialUsage
+var oboCredential = new OnBehalfOfCredential(clientId, clientSecret, userAccessToken);
+
+var client = new SecretClient(new Uri("https://myvault.vauls.azure.net/"), oboCredential);
+```
+
+## Rotating client certificates
```

Comment:
Is this meant to be here? Seems like this is all covered in the other sample md.